src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -332,22 +312,17 @@ else if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user))
     }
 
     /**
-     * GET /exercises/:exerciseId/toggle-second-correction
+     * POST exercises/:exerciseId/toggle-second-correction
      *
      * @param exerciseId the exerciseId of the exercise to get the repos from
      * @return the ResponseEntity with status 200 (OK) and with body the exercise, or with status 404 (Not Found)
      */
-    @PutMapping(value = "/exercises/{exerciseId}/toggle-second-correction")
+    @PostMapping("/exercises/{exerciseId}/toggle-second-correction")// TODO: should be courses/{courseId}/exercises/{exerciseId}/stats-for-assessment-dashboard
     @PreAuthorize("hasRole('INSTRUCTOR')")
     public ResponseEntity<Boolean> toggleSecondCorrectionEnabled(@PathVariable Long exerciseId) {
         log.debug("toggleSecondCorrectionEnabled for exercise with id: {}", exerciseId);
         Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
-        if (exercise == null) {
-            throw new EntityNotFoundException("Exercise not found with id " + exerciseId);
-        }
-        if (!authCheckService.isAtLeastInstructorForExercise(exercise)) {
-            return forbidden();
-        }
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.INSTRUCTOR, exercise, null);
```

Comment:
Does checkHasAtLeastRoleForExerciseElseThrow check if the exercise is present and throws a NotFound error when it is not?

Reply:
No it doesnt, but thats why all the exercises that are used for checking authority are directly fetched from the repository before and checked for existence